daemon/algod/api/server/v2/handlers.go

```diff
@@ -1111,6 +1113,33 @@ func (v2 *Handlers) GetApplicationByID(ctx echo.Context, applicationID uint64) e
 	return ctx.JSON(http.StatusOK, response)
 }
 
+// GetApplicationBoxByName returns the value of an application's box
+// (GET /v2/applications/{application-id}/boxes/{box-name})
+func (v2 *Handlers) GetApplicationBoxByName(ctx echo.Context, applicationID uint64, boxName string) error {
+	appIdx := basics.AppIndex(applicationID)
+	ledger := v2.Node.LedgerForAPI()
+
+	lastRound := ledger.Latest()
+
+	boxName, err := url.PathUnescape(boxName)
+	if err != nil {
+		return badRequest(ctx, err, err.Error(), v2.Log)
+	}
+	value, err := ledger.LookupKv(lastRound, logic.MakeBoxKey(appIdx, boxName))
+	if err != nil {
+		return internalError(ctx, err, errFailedLookingUpLedger, v2.Log)
+	}
+
+	if value == nil {
+		return notFound(ctx, errors.New(errBoxDoesNotExist), errAppDoesNotExist, v2.Log)
```

Comment:
@jannotti Request for clarification:  Can you help me understand why the internal and external errors differ here?

It's unclear to me why the _external_ message suggests the App does not exist.  As I understand it, it's possible either the app _or_ box do not exist.

If my understanding is accurate and provided it's not deemed too expensive, what do you think about checking for presence of App (via `GetApplicationByID`) to help provide a more specific error?

Alternatively, I think reusing `errBoxDoesNotExist` for the _external_ message feels close enough.

Reply:
typo

We shouldn't check if the app exists.  The box could continue to exist after an app is deleted.  It becomes a never ending memorial to the app's burned algos.